src/Corporation/OfficeSpace.ts

```diff
@@ -184,6 +184,7 @@ export class OfficeSpace {
       }
     }
 
+    if (jobCount == amount) return false;
     if ((jobCount + unassignedCount) < amount) return false;
```

Comment:
I think there's a fundamental problem with this entire function.

When someone calls e.g. `setEmployeeToJob(job, 2)` they're probably expecting that this will take two "unassigned" employees, and move them to this job. Implementing that would be easy.

But the way this function was designed, it will try to set employees such that the total number of employees assigned to this job is 2. So if there was already 1, it'll only add 1, and if there were already 2, it will add none. I think that's pretty weird, unless the function were named something like `setTotalNumberOfEmployeesOnJob`.

Anyway, not your problem, you're just fixing a bug (probably one of many with this method) but something to think about for another PR.

Reply:
nah 100% im a come back to this hopefully later today, i was just tired when the fix was requested 😆  and my brain said "yep that fixes x" but i could see that it really should just be self fixable, its something todo with missing an early out in the for loop shouldnt be too hard :)